### DISCUSSION TITLE
Conditional content: Website vs. Presentation

### DISCUSSION THREAD

**Discussion by johannesingrisch** [quarto-dev/quarto-cli#3215] [Q&A] [answered]
Hi all, 
is it possible to use conditional content not only for different output formats (e.g., html vs. pdf) but also for different document types?

Background: I am creating lecture slides (reveal js) and I want to automatically publish these as a website. On the website, I want to add additional content, which is not part of the lecture slides. 

Best,

**Comment by mcanouil** [comment, marked as answer]
This is what "project profiles" and "conditional content" features do (see <https://quarto.org/docs/projects/profiles.html> and <https://quarto.org/docs/authoring/conditional.html>).

To have access to those features, Quarto >= 1.2 is required.

**Reply by dragonstyle** [reply]
And further I don't believe that you are required to use the conditional content aliases - e.g. you can target `revealjs` for conditional content and that should work...

**Reply by johannesingrisch** [reply]
Both options work perfectly fine. Thanks!

**Reply by mcanouil** [reply]
@dragonstyle I have a follow-up question: how to literally targets only `html` since `html` is an alias for everything?

```
---
title: Title
format:
  revealjs: default
  html: default
---

::: {.content-visible when-format="html"}

Will only appear in HTML and not in Reveal.js, but actually it appears.

:::

::: {.content-visible when-format="revealjs"}

Will only appear in Reveal.js and not in HTML or other formats.

:::

```

**Reply by mcanouil** [reply]
I figure a way to do this, but it is definitely not straightforward at all.

I don't think alias should be named after formats.
In the documentation, there are formats with `*`, one might think of using those, but it does not work.
I would rather have an alias ends with `*` (e.g., `html*`) in its definition and to be able to use `html` to target the actual `html` format, not the type of format.

```
---
title: Title
format:
  revealjs: default
  html: default
---

::: {.content-visible when-format="html" unless-format="revealjs"}

Will only appear in HTML and not in Reveal.js, but actually it appears.

:::

::: {.content-visible when-format="revealjs"}

Will only appear in Reveal.js and not in HTML or other formats.

:::

```

**Reply by liyongg** [reply]
@mcanouil Is there also a way to do something like:
```
::: {.content-visible when-format="revealjs|beamer"}

This content is only displayed in RevealJS and/or Beamer output.

:::
```

Perhaps there is a workaround for this particular case, but I am also interested in a more general approach. Perhaps a feature request if this is not already possible?

**Reply by mcanouil** [reply]
Simply had another "when-format" statement.
No need to invent syntaxes.

Note that I don't think "beamer" exists as it's actually a PDF format.

**Reply by liyongg** [reply]
I see, thank you. I wasn't aware that that syntax would work as it was not directly clear to me from the docs. Would it be an idea to explicitly mention this and its reverse in the docs? The reverse (multiple `unless-format=""`) also seems to work and can be quite useful too.
